### PR TITLE
Support Amazon Simple Storage Service (AWS S3) metrics monitoring

### PR DESCRIPTION
**Pull request by pg-yang**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

Related issue https://github.com/apache/skywalking/issues/9883

### Screenshots

image

image

### CODE REVIEW

File: docs/en/setup/backend/backend-aws-s3-monitoring.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+[OpenTelemetry receiver](opentelemetry-receiver.md) and into the [Meter System](./../../concepts-and-designs/meter.md).
+
+### Data flow
+1. AWS CloudWatch collect metrics for S3, refer to [S3 monitoring with CloudWatch](https://docs.aws.amazon.com/AmazonS3/latest/userguide/cloudwatch-monitoring.html)
+2. [CloudWatch metric streams](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Metric-Streams.html) stream CloudWatch metrics of S3 to AWS Kinesis Data Firehose
+3. AWS Kinesis Data Firehose delivery metrics to [AWS Kinesis Data Firehose receiver](./aws-firehose-receiver.md) through the HTTP endpoint
+
+### Set up
+1. Create CloudWatch metrics configuration for S3, refer to [S3 metrics configuration](https://docs.aws.amazon.com/AmazonS3/latest/userguide/metrics-configurations.html)
+2. Create an Amazon Kinesis Data Firehose Delivery Stream, and set [AWS Kinesis Data Firehose receiver](./aws-firehose-receiver.md)'s address as HTTP Destination, refer to [Create Delivery Stream](https://docs.aws.amazon.com/firehose/latest/dev/basic-create.html)
```
Comment (wu-sheng) [inline docs/en/setup/backend/backend-aws-s3-monitoring.md:12]:
```suggestion
2. Create an Amazon Kinesis Data Firehose Delivery Stream, and set [AWS Kinesis Data Firehose receiver](./aws-firehose-receiver.md)'s address as HTTP(s) Destination, refer to [Create Delivery Stream](https://docs.aws.amazon.com/firehose/latest/dev/basic-create.html)
```

File: docs/en/setup/backend/opentelemetry-receiver.md [hunk 2]
```diff
@@ -43,5 +43,6 @@ for identification of the metric data.
 | Metrics of AWS Cloud EKS| otel-rules/aws-eks/eks-cluster.yaml |AWS Container Insights Receiver -> OpenTelemetry Collector -- OC/OTLP exporter --> SkyWalking OAP Server |
 | Metrics of AWS Cloud EKS| otel-rules/aws-eks/eks-service.yaml |AWS Container Insights Receiver -> OpenTelemetry Collector -- OC/OTLP exporter --> SkyWalking OAP Server |
 | Metrics of AWS Cloud EKS| otel-rules/aws-eks/eks-node.yaml |AWS Container Insights Receiver -> OpenTelemetry Collector -- OC/OTLP exporter --> SkyWalking OAP Server |
+| Metrics of AWS Cloud S3 | otel-rules/aws-s3/s3-service.yaml | AWS CloudWatcher Metrics Stream -> AWS Firehose delivery stream -> SkyWalking OAP Server with [AWS Firehose receiver](./aws-firehose-receiver.md) |
```
Comment (wu-sheng) [inline docs/en/setup/backend/opentelemetry-receiver.md:46]: I think we have mentioned this in `firehose receiver`, and should not mention here? At least, they are different protocol until AWS bumping up OTLP version.

Comment (pg-yang) [reply docs/en/setup/backend/opentelemetry-receiver.md:46]: Agree.  Actually, I forgot to remove this content

File: docs/menu.yml [hunk 3]
```diff
@@ -189,6 +189,8 @@ catalog:
             catalog:
               - name: "EKS Monitoring"
                 path: "/en/setup/backend/backend-aws-eks-monitoring"
+              - name: "S3 Monitoring"
+                path: "/en/setup/backend/backend-aws-s3-monitoring"
```
Comment (wu-sheng) [inline docs/menu.yml:193]: Just FYI, we plan to add DynamoDB on both AWS and Database menu(UI and docs)

https://github.com/apache/skywalking/pull/10380#discussion_r1103720431

File: oap-server/server-starter/src/main/resources/otel-rules/aws-s3/s3-service.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+#    "PT10H"     -- parses as "10 hours" (where an hour is 3600 seconds)
+#    "P2D"       -- parses as "2 days" (where a day is 24 hours or 86400 seconds)
+#    "P2DT3H4M"  -- parses as "2 days, 3 hours and 4 minutes"
+#    "P-6H3M"    -- parses as "-6 hours and +3 minutes"
+#    "-P6H3M"    -- parses as "-6 hours and -3 minutes"
+#    "-P-6H+3M"  -- parses as "+6 hours and -3 minutes"
+# </pre>
+
+filter: "{ tags -> {tags.cloud_provider == 'aws' && tags.Namespace == 'AWS/S3' } }"
+expSuffix: tag({tags -> tags.bucket = 'aws-s3-buckets::' + tags.BucketName}).service(['bucket'], Layer.AWS_S3)
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/otel-rules/aws-s3/s3-service.yaml:33]:
```suggestion
expSuffix: tag({tags -> tags.bucket = 'aws-s3::' + tags.BucketName}).service(['bucket'], Layer.AWS_S3)
```

As a group name, how about making it shorter?

File: oap-server/server-starter/src/main/resources/otel-rules/aws-s3/s3-service.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+metricPrefix: aws_s3
+metricsRules:
+  - name: 4xx
+    exp: amazonaws_com_AWS_S3_4xxErrors_sum.downsampling(SUM)
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/otel-rules/aws-s3/s3-service.yaml:37]: You mentioned this as a count in the document. Are you meaning 4xx requests count in the current time bucket? (minute/hour/day)

**Comment by wu-sheng** [comment]
I think we should polish the dashboard a little.
* Add 4 single values as the first line. 4xx, 5xx, Request Count, Request Latency
* 4xx, 5xx, Request latency, first byte, as line two
* 4 counts for all, get, put and delete as line three
* Download and Uploaded at line four, one for half of the line, also open this new option
image

File: oap-server/server-starter/src/main/resources/otel-rules/aws-s3/s3-service.yaml (hunk 5 above)
Comment (wankai123) [reply oap-server/server-starter/src/main/resources/otel-rules/aws-s3/s3-service.yaml:37]: From the AWS doc: 
`Sum (reports per period)`, sorry I didn't found the period‘s definition, could you share here?
I think downsampling(SUM) could be a issue.

```
4xxErrors:
The number of HTTP 4xx client error status code requests made to an Amazon S3 bucket with a value of either 0 or 1. The Average statistic shows the error rate, and the Sum statistic shows the count of that type of error, during each period.

Units: Count

Valid statistics: Average (reports per request), Sum (reports per period), Min, Max, Sample Count
```

Comment (wu-sheng) [reply oap-server/server-starter/src/main/resources/otel-rules/aws-s3/s3-service.yaml:37]: I think this is a report period, then, we are doing summary for data in our dimensions.
If it is reported per several seconds, we are still good.

Comment (pg-yang) [reply oap-server/server-starter/src/main/resources/otel-rules/aws-s3/s3-service.yaml:37]: From doc [Monitoring metrics with Amazon CloudWatch](https://docs.aws.amazon.com/pdfs/AmazonS3/latest/userguide/s3-userguide.pdf#cloudwatch-monitoring), `Request Metrics` are available at 1-minute intervals. I think it means the reporting period is 1 minute and I have confirmed it from logs

Comment (wu-sheng) [reply oap-server/server-starter/src/main/resources/otel-rules/aws-s3/s3-service.yaml:37]: I think generally, this should be good.

**Comment by pg-yang** [comment]
Update UI :

image

**Review by wu-sheng — APPROVED** [review]
LGTM.

**Pull request merged**